src/libs/common/ide/types/ide.types.ts

```diff
@@ -173,6 +176,28 @@ export interface IDE {
    * `undefined` when the command handler function doesn't return anything.
    */
   executeCommand<T>(command: string, ...args: any[]): Promise<T | undefined>;
+
+  /**
+   * Temporarily emphasize the given ranges to the user.  This function is used
+   * to show ranges that eg are about to be deleted, are the source of a bring,
+   * etc. The promise should resolve when the flash is complete.
+   *
+   * @param flashDescriptors Descriptions of the ranges to flash, including
+   * information about the type of flash
+   */
+  flashRanges(flashDescriptors: FlashDescriptor[]): Promise<void>;
+
+  /**
+   * Set the ranges to which {@link highlightId} should be applied.  Removes
+   * the given highlight from all other ranges in all editors.
+   *
+   * @param highlightId The id of the highlight to apply
+   * @param ranges The ranges to apply the highlight to
+   */
+  setHighlightRanges(
```

Comment:
file issue to migrate this to `TextEditor`, as we don't want to remove highlights from other editors when we set in one editor, so this api seems confusing

Reply:
for now just add `editor` param and have list of generalised ranges (without editors) instead of `EditorGeneralizedRange`